clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -204,23 +204,14 @@ public synchronized void ensureCoordinatorReady() {
                 time.sleep(retryBackoffMs);
             }
         }
+        // Since coordinator is set in a synchronized block in the group coordinator response callback,
+        // the future associated with the cooordinator request can be safely cleared here.
+        findCoordinatorFuture = null;
```

Comment:
@ijuma It was intentional. I should have clarified this with a comment, but I was concerned about calling `lookupCoordinator()` in a tight loop without the ability to backoff. The natural thing would probably be to call `lookupCoordinator()` and then poll on the future, but one of the super awkward things with sharing the NetworkClient with the heartbeat thread is keeping it from blocking the main thread. I think basically this is telling us that we shouldn't be doing this, so longer term I'd like to look into moving all coordinator comms into the background thread and giving it its own NetworkClient. For now, if you can think of better ways to structure this, I'd love to hear them.

By the way, using try/finally makes sense, but we have to be a little careful since poll() can be woken up before the request has returned. It should probably only wrap the calls after the poll().